### DISCUSSION TITLE
Extracting Normalized Integrated Counts for External Analysis

### DISCUSSION THREAD

**Discussion by RustemKurkcu** [satijalab/seurat#8413] [Q&A]
Hi, 

 I am currently working with a dataset consisting of six treatment groups and a control, and I've been following the 
 integration and normalization guides provided in the Seurat documentation (Introduction to Integration and Seurat v5
 Integration).

My primary goal is to analyze normalized integrated counts outside of Seurat, specifically for generating heatmaps 
and exploring correlations with external tools. However, I've encountered challenges in extracting these counts in a
 format similar to what DESeq2 provides, which would be ideal for my analysis needs.

**Could anyone advise on the following points:**

Extracting Normalized Integrated Counts: Is there a direct method to extract normalized integrated counts from Seurat 
objects? If so, could you please share the appropriate commands or functions?

Recommendations for External Analysis: 
If direct extraction isn't feasible, are there recommended workflows or 
practices for creating an integrated counts matrix that can be analyzed outside of Seurat, similar to DESeq2's approach?

Optimal Integration and Normalization Methods for External Use:
 Given my dataset's structure (6 treatments + control),
 which integration and normalization methods within Seurat would you recommend for subsequent external analysis?

I've attempted multiple approaches based on the tutorials but would greatly appreciate guidance on best practices, 
especially for preparing data for external heatmap and correlation analyses.

Sorry to ask, from what I gather there is no direct way,

**Comment by patrickCNMartin** [comment]
Bumping the thread for interest!

I have been looking into the same thing.  Since I am not Seurat developer and just I user, I would suggest taking my suggestion with a grain of salt but:

Getting the integrated counts is straight forward but they are scaled counts which might not be the most appropriate for your use case. You can find them at `obj@assay$RNA@layers`. They will be called whatever you called put as `new.reduction` in the `IntegrateLayers` function. It is should be the last layer added. The `RNA` assay is the default one but use which ever one you have been using for your integration. 

To get the normalized counts, I have tried using the cell embeddings and gene loadings of the integrated  reduction. Something along the lines of:

```
# Using your integrated layer object where the integrated reduction is called 'integrated'
counts <- obj@assay$RNA$layers
integrate <- t(as.matrix(cbind(counts[["data.1"]], counts[["data.2"]])))
integrate <- integrate[colnames(counts[["integrated"]]),
        rownames(counts[["integrated"]])]
mu <- colMeans(integrate)
back <- embeds@cell.embeddings %*% t(embeds@feature.loadings)
back <- t(scale(back, center = -mu, scale = FALSE))
back[which(back <= 0, arr.ind = TRUE)] <- 0
    
 ```
 This more or less approximates the log norm counts from the integrated latent space. 
 
 
 I am not sure this is the most appropriate method hence I would be very happy to hear what the community has to suggest!
